File: negocio-tundama-backend/src/main/java/com/tundama/negociotundamabackend/model/entity/Producto.java
package com.tundama.negociotundamabackend.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore; // IMPORTAR ESTA

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "productos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "id_producto", unique = true, nullable = false, length = 50)
    private String idProducto;

    @Column(name = "nombre", nullable = false, length = 100)
    private String nombre;

    @Column(name = "descripcion", length = 500)
    private String descripcion;

    @Column(name = "unidad_medida", length = 20)
    private String unidadMedida;

    @Column(name = "precio_unitario", precision = 10, scale = 2)
    private BigDecimal precioUnitario;

    @Column(name = "iva", precision = 5, scale = 2)
    private BigDecimal iva;

    @Column(name = "stock_actual")
    private Integer stockActual = 0;

    @Column(name = "stock_minimo")
    private Integer stockMinimo = 0;

    // ============= AGREGAR @JsonIgnore AQUÍ (opcional) =============
    @JsonIgnore  // Si no quieres que aparezca el proveedor en la respuesta JSON de productos
    @ManyToOne
    @JoinColumn(name = "proveedor_id")
    private Proveedor proveedor;
    // ===============================================================

    @Column(name = "fecha_registro")
    private LocalDateTime fechaRegistro;

    @Column(name = "activo")
    private Boolean activo = true;

    @PrePersist
    public void prePersist() {
        fechaRegistro = LocalDateTime.now();
    }
}

File: Version Android/lib/models/producto.dart
// ==================== lib/models/producto.dart ====================
class Producto {
  final int? id;
  final String idProducto;
  final String nombre;
  final String? descripcion;
  final String? unidadMedida;
  final double? precioUnitario;
  final double? iva;
  final int stockActual;
  final int stockMinimo;
  final int? proveedorId;
  final String? proveedorNombre;
  final DateTime? fechaRegistro;
  final bool activo;

  Producto({
    this.id,
    required this.idProducto,
    required this.nombre,
    this.descripcion,
    this.unidadMedida,
    this.precioUnitario,
    this.iva,
    this.stockActual = 0,
    this.stockMinimo = 0,
    this.proveedorId,
    this.proveedorNombre,
    this.fechaRegistro,
    this.activo = true,
  });

  factory Producto.fromJson(Map<String, dynamic> json) {
    return Producto(
      id: json['id'],
      idProducto: json['idProducto'] ?? '',
      nombre: json['nombre'] ?? '',
      descripcion: json['descripcion'],
      unidadMedida: json['unidadMedida'],
      precioUnitario: (json['precioUnitario'] ?? 0).toDouble(),
      iva: (json['iva'] ?? 0).toDouble(),
      stockActual: json['stockActual'] ?? 0,
      stockMinimo: json['stockMinimo'] ?? 0,
      proveedorId: json['proveedor'] != null ? json['proveedor']['id'] : null,
      proveedorNombre: json['proveedor'] != null ? json['proveedor']['nombreEmpresa'] : null,
      fechaRegistro: json['fechaRegistro'] != null 
          ? DateTime.parse(json['fechaRegistro']) 
          : null,
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'descripcion': descripcion,
      'unidadMedida': unidadMedida,
      'precioUnitario': precioUnitario,
      'iva': iva,
      'stockActual': stockActual,
      'stockMinimo': stockMinimo,
    };
  }

  bool get isLowStock => stockActual <= stockMinimo;
}
